Formula/ipopt.rb

```diff
@@ -0,0 +1,60 @@
+class Ipopt < Formula
+  desc "Interior point optimizer"
+  homepage "https://projects.coin-or.org/Ipopt/"
+  url "https://www.coin-or.org/download/source/Ipopt/Ipopt-3.12.12.tgz"
+  sha256 "7baeb713ef8d1999bed397b938e9654b38ad536406634384455372dd7e4ed61f"
+  head "https://github.com/coin-or/Ipopt.git"
+
+  depends_on "gcc"
+
+  resource "coinmumps" do
+    # https://github.com/coin-or-tools/ThirdParty-Mumps/blob/35dba763084f6f9ed4db087e2e5e53e80a92bb15/get.Mumps#L31
+    url "http://mumps.enseeiht.fr/MUMPS_4.10.0.tar.gz"
+    sha256 "d0f86f91a74c51a17a2ff1be9c9cee2338976f13a6d00896ba5b43a5ca05d933"
+  end
+
+  def install
+    ENV.delete("MPICC")
+    ENV.delete("MPICXX")
+    ENV.delete("MPIFC")
+
+    resource("coinmumps").stage do
+      # https://github.com/coin-or-tools/ThirdParty-Mumps/blob/35dba763084f6f9ed4db087e2e5e53e80a92bb15/get.Mumps#L51-L53
+      system "patch", "-p1", "-i", buildpath/"ThirdParty/Mumps/mumps.patch"
```

Comment:
So it actually needs a patched, old and unmaintained version? Not sure if that's cool for Homebrew core. What about maintaining the whole thing in a tap?

Reply:
Trying to migrate from a tap (one of the many for this exact formula, which is why, technical issues aside, it feels like it should be in core).

It doesn't actually need an old MUMPS, but the build files are written for that version. We could use a new MUMPS, but then we have to deal with the MUMPS build files. Let me see if that is viable.

Reply:
@fxcoudert PTAL. This uses the latest MUMPS.